x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -111,21 +175,29 @@ export async function getDocumentSources({
       rollupInterval,
       hasDataBefore,
       hasDataAfter,
+      checkSummaryExistsField: source.meta.checkSummaryExistsField,
     };
   });
 
-  const hasAnyDataBefore = checkedSources.some(
-    (source) => source.hasDataBefore
-  );
+  const hasAnyDataBefore = checkedSources
+    .filter((source) => !source.checkSummaryExistsField)
+    .some((source) => source.hasDataBefore);
 
-  const sources: Array<ApmDataSource & { hasDocs: boolean }> =
-    checkedSources.map((source) => {
-      const { documentType, hasDataAfter, hasDataBefore, rollupInterval } =
-        source;
+  const sources: TimeRangeMetadata['sources'] = [];
 
-      const hasData = hasDataBefore || hasDataAfter;
+  checkedSources.forEach((source) => {
+    const {
+      documentType,
+      hasDataAfter,
+      hasDataBefore,
+      rollupInterval,
+      checkSummaryExistsField,
+    } = source;
+    const hasData = hasDataBefore || hasDataAfter;
+    const hasDocsData = hasAnyDataBefore ? hasDataBefore : hasData;
```

Comment:
There's probably a better way to name this. Can you make the variables more descriptive? This is an important piece of code so I think it is rewarding if we're more specific/accurate here.

Reply:
@dgieselaar I am somehow unable to reply to your original comment so replying it here.
> why do have to exclude hasDurationSummary searches?

The check `hasAnyDataBefore` checks for documents to be present based on a different expectations (es queries) as against the ones we are doing for `hasDurationSummary` which only checks for the field to be present or not. This is why in order not to cause any unknown side effects i filtered them out.